### PR TITLE
Refactor interface lowering

### PR DESCRIPTION
**Pull request by aykevl**
This is something I've wanted to do for a long time, but couldn't do before doing other refactors (in particular: https://github.com/tinygo-org/tinygo/pull/677). It should not change any behavior, only move code around.

The first commit adds a llvmutil package. I couldn't think of a better way to do it: the code that's moved is hard to get right and should really not be copied. If you know of a better way, please let me know. Also, please let me know if you can think of a better package name/location.

Tests will be added soon, submitting now for initial review.

### CODE REVIEW

**Comment by aykevl** [comment]
And now with tests.

File: compiler/llvm.go [hunk 1]
```diff
@@ -24,79 +23,33 @@ func getUses(value llvm.Value) []llvm.Value {
...
 func (c *Compiler) createTemporaryAlloca(t llvm.Type, name string) (alloca, bitcast, size llvm.Value) {
-	alloca = c.createEntryBlockAlloca(t, name)
-	bitcast = c.builder.CreateBitCast(alloca, c.i8ptrType, name+".bitcast")
-	size = llvm.ConstInt(c.ctx.Int64Type(), c.targetData.TypeAllocSize(t), false)
-	c.builder.CreateCall(c.getLifetimeStartFunc(), []llvm.Value{size, bitcast}, "")
-	return
-}
-
-// createInstructionAlloca creates an alloca in the entry block, and places lifetime control intrinsics around the instruction
-func (c *Compiler) createInstructionAlloca(t llvm.Type, inst llvm.Value, name string) llvm.Value {
```
Comment (jadr2ddude) [inline compiler/llvm.go]: Is there a reason that this is handled differently than the rest?

Comment (aykevl) [reply compiler/llvm.go]: The only reason is that `createInstructionAlloca` was only called once so I just modified the caller.
That's not a very good reason, and I could certainly add a wrapper here. No real preference by me either way.

Comment (jadr2ddude) [reply compiler/llvm.go]: idk, doesnt make too much of a difference

**Pull request merged**